nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/RetrieveFromColdStorage.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.collectors.DocumentModelCollector;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+
+/**
+ * Retrieves the cold storage content associated with the input {@link DocumentModel}. This operation will initiate a
+ * restore request, calling the {@link Blob#getStream()} during this process doesn't mean you will get the blob's
+ * content.
+ * 
+ * @since 11.1
+ */
+@Operation(id = RetrieveFromColdStorage.ID, category = Constants.CAT_BLOB, label = "Request restoration from cold storage", description = "Request a cold storage content restoration associated with the document.")
+public class RetrieveFromColdStorage {
+
+    public static final String ID = "Document.RetrieveFromColdStorage";
```

Comment:
I feel that the operation name and classname (and other method names) is misleading. People will expect "retrieve" to return the blob and not be async. AWS S3 calls it "restore". But this is in fact a request to restore the blob at a later point in time.

Reply:
yes, please see https://github.com/nuxeo/nuxeo/pull/3666#discussion_r375140959 and if we want we change the jira ticket/validate with the PO to rename it.